src/com/facebook/buck/apple/AppleLibraryDescription.java

```diff
@@ -769,6 +828,68 @@ private boolean shouldWrapIntoDebuggableBinary(BuildTarget buildTarget, BuildRul
     return Optional.empty();
   }
 
+  private <U> Optional<U> createCxxPreprocessorInputMetadata(
+      BuildTarget buildTarget,
+      BuildRuleResolver resolver,
+      CellPathResolver cellRoots,
+      AppleNativeTargetDescriptionArg args,
+      Class<U> metadataClass,
+      Entry<Flavor, CxxLibraryDescription.MetadataType> cxxMetaDataType) {
+    Entry<Flavor, CxxPlatform> platform =
+        getCxxPlatformsProvider()
+            .getCxxPlatforms()
+            .getFlavorAndValue(buildTarget)
+            .orElseThrow(IllegalArgumentException::new);
+    Entry<Flavor, HeaderVisibility> visibility =
+        CxxLibraryDescription.HEADER_VISIBILITY
+            .getFlavorAndValue(buildTarget)
+            .orElseThrow(IllegalArgumentException::new);
+    BuildTarget baseTarget =
+        buildTarget.withoutFlavors(
+            cxxMetaDataType.getKey(), platform.getKey(), visibility.getKey());
+
+    CxxPreprocessorInput.Builder cxxPreprocessorInputBuilder = CxxPreprocessorInput.builder();
```

Comment:
@robbertvanginkel Seems like the code from 851-862 is duplicated from `CxxLibraryMetadataFactory`. Can we just extract and expose it instead?

Reply:
Refactored into a static method on `CxxLibraryMetadataFactory`